src/Components/ObjectsModalTable/ObjectsModalTable.js

```diff
@@ -0,0 +1,207 @@
+  const perPage = filters.limit;
+
+  const removeFilterParam = (param) =>
+    _removeFilterParam(filters, updateFilters, param);
+
+  const updateFilters = (payload) =>
+    dispatch(updateWorkloadsObjectsListFilters(payload));
+  const preparedRows = displayedRows.length > 0 ? true : false;
+  const loadingState = !rowsFiltered;
+
+  const filterConfigItems = [
+    {
+      label: 'Object ID',
+      filterValues: {
+        key: 'object_id',
+        onChange: (_event, value) =>
+          updateFilters({ ...filters, offset: 0, object_id: value }),
+        value: filters.object_id,
+        placeholder: 'Filter by Object ID',
+      },
+    },
+  ];
+
+  const buildFilterChips = () => {
+    const localFilters = { ...filters };
+    delete localFilters.sortIndex;
+    delete localFilters.sortDirection;
+    return pruneWorkloadsRulesFilters(localFilters, {
+      label: 'Object ID',
+      type: 'text',
+      title: 'object ID',
+      urlParam: 'object_id',
+    });
+  };
+
+  const activeFiltersConfig = {
+    showDeleteButton: filtersApplied,
+    deleteTitle: 'Reset filters',
+    filters: buildFilterChips(),
+    onDelete: (_event, itemsToRemove, isAll) => {
+      if (isAll) {
+        resetFilters(
+          filters,
+          WORKLOADS_OBJECTS_TABLE_INITIAL_STATE,
+          updateFilters
+        );
+      } else {
+        itemsToRemove.map((item) => {
+          const newFilter = {
+            [item.urlParam]: Array.isArray(filters[item.urlParam])
+              ? filters[item.urlParam].filter(
+                  (value) => String(value) !== String(item.chips[0].value)
+                )
+              : '',
+          };
+          newFilter[item.urlParam].length > 0
+            ? updateFilters({ ...filters, ...newFilter })
+            : removeFilterParam(item.urlParam);
+        });
+      }
+    },
+  };
+
+  //This is where we apply filters and map rows agains the filters
+  const buildFilteredRows = (allrows, filters) => {
+    return allrows.filter((object) => passObjectsFilters(object, filters));
+  };
+
+  //building displayed rows applying limits and per page values
+  const buildDisplayedRows = (rows) => {
+    return rows.slice(perPage * (page - 1), perPage * (page - 1) + perPage);
+  };
+
+  //After objectsData is present or in case of object id filter change we setFiltered rows using buildiflterRows
+  useEffect(() => {
+    setFilteredRows(buildFilteredRows(objectsData, filters));
+  }, [objectsData, filters]);
+
+  //after objects data is present we set filtered rows and this useEffect is triggered to update displayed rows
+  //with new array of rows that have filters applied
+  useEffect(() => {
+    setDisplayedRows(buildDisplayedRows(filteredRows));
+    setFiltersApplied(filtersAreApplied(filters));
+    setRowsFiltered(true);
+  }, [filteredRows, filters.limit, filters.offset]);
+
+  const onSetPage = (_e, pageNumber) => {
+    setRowsFiltered(false);
```

Comment:
Will pagination and sorting handle the backend? If not we can remove this to prevent the unnecessary flickering.

Reply:
Yep, removed it